### PR TITLE
Fixed bug for navData.json generation

### LINKED ISSUE #337: Refactor _createNavData and SideNav to avoid dashedName collisions
I'm currently taking a course on Data Structures and I think a section on this subject would be very useful. Content could include: Bags, Dictionaries, Sets, Trees, etc. 
I'm currently implementing these structures using Java; however, we could provide multiple implementations in different languages such as Java, Python, C++, and even JavaScript.

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
navData used to use the `dashedName` as page keys. Now uses `path` instead.
Closes #337

### CODE REVIEW

File: src/LayoutComponents/nav/SideNav.jsx [hunk 1]
```diff
@@ -32,44 +32,40 @@ function mapDispatchToProps(dispatch) {
...
-          title={ child.title }
-        />
-      );
+  return children.map(child => {
+    if (child.hasChildren) {
+      return renderParent(child, pages);
+    }
+    return (
+      <NavItem
+        isStubbed={child.isStubbed}
```
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: I think my Atom-Prettier plugin accidentally formatted this file. I'll turn it off and resubmit the changes!

File: src/LayoutComponents/nav/SideNav.jsx [hunk 2]
```diff
@@ -32,44 +32,40 @@ function mapDispatchToProps(dispatch) {
...
-    const children = renderChildren(childrenForParent, pages);
+  const children = renderChildren(childrenForParent, pages);
 
   return (
-    <NavPanel
-      key={ parent.path }
-      path={ parent.path }
-      >
-      { children }
+    <NavPanel key={parent.path} path={parent.path}>
```
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: Atom-Prettier strikes again

File: src/LayoutComponents/nav/SideNav.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: When you say spaces around curly braces do you mean 
```javascript
<Comp prop = {property} />
```
or
```javascript
<Comp prop={ property }/>
```
Or something entirely different?

File: src/LayoutComponents/nav/SideNav.jsx [hunk 3]
```diff
@@ -79,11 +75,13 @@ class SideNav extends Component {
...
-    const pathMap = pathname.slice(1).split('/').slice(0, -1)
+    const pathMap = pathname
+      .slice(1)
+      .split('/')
+      .slice(0, -1)
       .reduce((accu, current, i, pathArray) => {
-        const path = i !== 0 ?
-          accu[pathArray[ i - 1 ]] + `/${current}` :
-          `/${current}`;
+        const path =
```
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: Atom-prettier's fault

File: seed/_createNavData.js [hunk 4]
```diff
@@ -10,47 +10,46 @@ const { commonREs, excludedDirs } = require('./utils');
...
-        parentPath: level.slice(9).replace(new RegExp(parent + '/'), ''),
-        title
-      };
-      if (!subDirs.length) {
-        // no child directories
-        return Observable.of(null);
-      }
-      return listAllDirs(thisDir, accuPages);
-    });
+    navData[thisDir.slice(9)] = {
```
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply seed/_createNavData.js]: Makes sense. :+1:

**Comment by Ethan-Arrowood** [comment]
Thank you for reviewing my code @systimotic! Yes I jumped the gun on giving up on this; it was like a eureka moment the second I posted the comment - for the future I'm always going to give it one more try before I hand off an issue. I'm also glad you enjoyed the livestream! I had so much fun recording it; I'm going to be producing more content soon on React, Node, & Python!

File: src/LayoutComponents/nav/SideNav.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/LayoutComponents/nav/SideNav.jsx]:
```jsx
<Comp spacedCurlies={ true } />
```

File: src/LayoutComponents/nav/SideNav.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/LayoutComponents/nav/SideNav.jsx]: I keep hearing about `prettier` maybe one for this project would be good. Issue in-bound

File: src/LayoutComponents/nav/SideNav.jsx [hunk 5]
```diff
@@ -104,11 +102,11 @@ class SideNav extends Component {
     return (
       <div className='sideNav' id='side-nav'>
         <PanelGroup>
-          {
-            (!parents || !expandedState) ?
-            <NavPanel title={'No Parents Here'}/> :
+          {!parents || !expandedState ? (
```
Comment (Bouncey) [inline src/LayoutComponents/nav/SideNav.jsx]: Is this a prettier thing also?

File: src/LayoutComponents/nav/SideNav.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: It has a plugin for atom that whenever you save a file it automatically 'prettifies' it. Its awesome!

File: src/LayoutComponents/nav/SideNav.jsx (hunk 5 above)
Comment (Ethan-Arrowood) [reply src/LayoutComponents/nav/SideNav.jsx]: Yes. I'll go fix this now too

**Comment by Ethan-Arrowood** [comment]
@systimotic & @Bouncey I think I've fixed all the changes you've requested. Let me know if theres anything else I need to do.

**Comment by systimotic** [comment]
Excited for more of your content @Ethan-Arrowood! This PR looks almost good to me, just one thing left. It looks like you (probably accidentally 😄) removed the entire `yarn.lock` with the first commit from today. Once that's restored, I think this is good to merge 👍

**Comment by Ethan-Arrowood** [comment]
@systimotic I don't seem to have the file in my local dir anymore. How would I regenerate it?

**Comment by systimotic** [comment]
@Ethan-Arrowood It should be generated automatically after running `yarn install`

**Comment by Ethan-Arrowood** [comment]
@systimotic okay, but this branch is currently having issues with `yarn install` due to the gyp thing. I've updated my master branch with `git pull --rebase upstream master`, how would I update this branch now? (Sorry about these silly git questions. I'm still trying to master git).

**Comment by Bouncey** [comment]
@Ethan-Arrowood `git push origin Fix/Refactor_createNavData -f`

**Comment by Ethan-Arrowood** [comment]
Alright I found a few minutes to review difference between merging & rebasing and I was able to update my branch, then push the missing yarn.lock file :+1:

**Comment by Bouncey** [comment]
## LGTM

Thanks for the effort @Ethan-Arrowood 

:sparkles: :sparkles: :sparkles: :sparkles:

**Pull request merged**